core/src/test/scala/org/apache/spark/shuffle/sort/SortShuffleWriterSuite.scala

```diff
@@ -90,9 +93,8 @@ class SortShuffleWriterSuite extends SparkFunSuite with SharedSparkContext with
     writer.stop(success = true)
     val dataFile = shuffleBlockResolver.getDataFile(shuffleId, 2)
     assert(dataFile.exists())
-    assert(dataFile.length() !== 0)
-    assert(context.taskMetrics().shuffleWriteMetrics.bytesWritten !== 0)
-    assert(context.taskMetrics().shuffleWriteMetrics.recordsWritten !== 0)
-    Utils.deleteRecursively(dataFile)
+    assert(dataFile.length() !== expected)
```

Comment:
Use shuffle write metrics, like:

```scala
    val writeMetrics = context.taskMetrics().shuffleWriteMetrics
    assert(dataFile.length() === writeMetrics.bytesWritten)
    assert(context.taskMetrics().shuffleWriteMetrics.bytesWritten === writeMetrics.bytesWritten)
    assert(context.taskMetrics().shuffleWriteMetrics.recordsWritten === writeMetrics.recordsWritten)
```

Reply:
`assert(dataFile.length() !== 0)` is too conservative. It would be great to compare `length()` with the specific value (e.g. as @attilapiros suggestion).